src/img-upload/imgUpload.service.ts

```diff
@@ -0,0 +1,51 @@
+import { Injectable, Req, Res } from '@nestjs/common';
+import * as multer from 'multer';
+import * as multerS3 from 'multer-s3';
+import { S3Client } from '@aws-sdk/client-s3';
+
+@Injectable()
+export class ImgUploadService {
+  private upload;
+
+  constructor() {
+    this.upload = multer({
+      storage: multerS3({
+        s3: new S3Client({
+          region: process.env.AWS_REGION,
+          credentials: {
+            accessKeyId: process.env.AWS_ACCESS_KEY_ID,
+            secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+          },
+        }),
+        contentType: multerS3.AUTO_CONTENT_TYPE,
+        bucket: process.env.AWS_BUCKET_NAME,
+        key: (request, file, cb) => {
+          cb(null, `image/main/${Date.now().toString()}-${file.originalname}`);
+        },
+      }),
+      limits: { fileSize: 5 * 1024 * 1024 },
+      fileFilter: (req, file, cb) => {
+        const filetypes = /jpeg|jpg|png|gif/;
+        const mimetype = filetypes.test(file.mimetype);
+        const extname = filetypes.test(file.originalname.split('.').pop().toLowerCase());
```

Comment:
pop보다는 인덱스로 접근해주는게 조금 더 일반적인 것 같아요! 
```suggestion
        const extname = filetypes.test(file.originalname.split('.')[1].toLowerCase());
```